cpu/stm32/periph/dma.c

```diff
@@ -193,22 +193,25 @@ static IRQn_Type dma_get_irqn(int stream)
         return ((IRQn_Type)((int)DMA1_Channel1_IRQn + stream));
     }
 #if defined(DMA2_BASE)
+    else if (stream == 7) {
+        return -1;
+    }
 #if defined(CPU_FAM_STM32F1)
     else if (stream < 11) {
 #else
     else if (stream < 13 ) {
 #endif
-        return ((IRQn_Type)((int)DMA2_Channel1_IRQn + stream));
+        return ((IRQn_Type)((int)DMA2_Channel1_IRQn + stream - 8));
     }
-#if !defined(CPU_FAM_STM32L1)
+#if !defined(CPU_FAM_STM32L1) && !defined(CPU_FAM_STM32F3)
     else {
 #if defined(CPU_FAM_STM32F1)
         return (DMA2_Channel4_5_IRQn);
 #else
-        return ((IRQn_Type)((int)DMA2_Channel6_IRQn + stream));
+        return ((IRQn_Type)((int)DMA2_Channel6_IRQn + stream - 13));
```

Comment:
Would be good to have a comment as to where those magic numbers come from.
When someone will touch this code later they will have no idea.

Reply:
Would this be sufficient? I don't really know how to explain the magic numbers without a very long description. They are already explained in the header file. They are also used heavily throughout the code in many places I did not touch. Adding to the difficulty in explaining them is that ST uses DMA "channels" (that are 1 to n based) in some parts, and "streams" (that are 0 to n based). Even worse the parts that use "streams" further break the streams in to "channels." So not easy to describe other than to maybe point people to the header file to decode the numbers?

```suggestion
        return ((IRQn_Type)((int)DMA2_Channel6_IRQn + stream - 13)); // 13 = 8 (DMA2) + 5 (Channel6)
```

Reply:
Yea that's already better than no comment at all.

Reply:
done

Reply:
Looks like you forgot to push

Reply:
Whoops, pushed to the wrong remote. Fixed now.